lib/openhab/core/items/semantics.rb

```diff
@@ -466,23 +470,25 @@ def self.add(label: nil, synonyms: "", description: "", **tags)
         #
         def self.lookup(id, locale = nil)
           id = id.to_sym
-          return const_get(id) if constants.include?(id)
+          return const_get(id) if constants.include?(id) || const_missing(id)
 
-          id = id.to_s
           locale = java.util.Locale.default if locale.nil?
-          tag = org.openhab.core.semantics.SemanticTags.get_by_id(id) ||
-                org.openhab.core.semantics.SemanticTags.get_by_label_or_synonym(id, locale).first
-          tag&.ruby_class
+          org.openhab.core.semantics.SemanticTags.get_by_label_or_synonym(id.to_s, locale).first&.ruby_class
         end
 
         #
         # Automatically looks up new semantic classes and adds them as `constants`
         #
         # @return [Tag, nil]
         #
+        # @!visibility private
         def self.const_missing(sym)
           logger.trace("const missing, performing Semantics Lookup for: #{sym}")
-          org.openhab.core.semantics.SemanticTags.get_by_id(sym.to_s)&.then { |tag| const_set(sym, tag.ruby_class) }
+          # @deprecated OH3.4 - the Property tag had an ID of "MeasurementProperty" in OH3.4. This was corrected in OH4.
+          sym = :MeasurementProperty if sym == :Property && Core::VERSION.split(".").first.to_i < 4
```

Comment:
use Gem::Version here instead of splitting yourself

Reply:
Thanks!